lib/beaker/dsl/install_utils/foss_utils.rb

```diff
@@ -245,21 +245,23 @@ def install_puppet(opts = {})
         # @option opts [String] :puppet_agent_version Version of puppet agent to download
         # @option opts [String] :mac_download_url Url to download msi pattern of %url%/puppet-%version%.msi
         # @option opts [String] :win_download_url Url to download dmg pattern of %url%/(puppet|hiera|facter)-%version%.msi
+        # @option opts [Boolean] :install_in_parallel   Whether to install on all hosts in parallel. Defaults to false.
         #
         # @return nil
         # @raise [StandardError] When encountering an unsupported platform by default, or if gem cannot be found when default_action => 'gem_install'
         # @raise [FailTest] When error occurs during the actual installation process
         def install_puppet_on(hosts, opts={})
+          run_in_parallel = opts[:install_in_parallel]
           opts = FOSS_DEFAULT_DOWNLOAD_URLS.merge(opts)
 
           # If version isn't specified assume the latest in the 3.x series
           if opts[:version] and not version_is_less(opts[:version], '4.0.0')
             # backwards compatability
             opts[:puppet_agent_version] ||= opts[:version]
-            install_puppet_agent_on(hosts, opts)
+            install_puppet_agent_on(hosts, run_in_parallel, opts)
 
           else
-            block_on hosts do |host|
+            block_on hosts, run_in_parallel do |host|
               if host['platform'] =~ /el-(5|6|7)/
                 relver = $1
                 install_puppet_from_rpm_on(host, opts.merge(:release => relver, :family => 'el'))
```

Comment:
Hmm... I have not exhaustively tested this, but I would expect in this case - or any case where it would otherwise cause things to blow up - you would opt not to run in parallel.  This would likely require some experimentation to determine whether it would work or not.